### ISSUE TITLE
Event t0

### ISSUE THREAD

**Issue by YifanC** [DUNE/ndlar_flow#109]
When we use `SymmetricWindowRawEventBuilder `, if there's no external trigger, the time of the first arrival hit is considered as event t0, which can be horrendously wrong.
1. Shall we explicitly label events without external triggers downstream from the flow stage?
2. It might not be good idea, but for cathode crossing events, shall we try stitching the hits at the cathode and consider that as the t0?
<img width="567" alt="evt7" src="https://github.com/DUNE/ndlar_flow/assets/25907864/a91c5f0f-e66a-4f71-948c-6ef02193f638">
<img width="551" alt="evt10" src="https://github.com/DUNE/ndlar_flow/assets/25907864/ba5f805a-71a5-49b1-858e-74a44642207e">

**Comment by krwood** [comment]
Yes, in general we should add a field to the charge/event object that tags which event builder built the particular event. Then for 2x2 we first run over with the beam trigger event builder, make all the events we can, and tag them as beam trigger events. And then, e.g., run over the remaining packets and build as many LRS threshold trigger events as we can, and tag them as threshold trigger events. And then another pass with something like the symmetric window event builder, and tag them as such. Etc.

Side note: for events like the ones you have in the attached event displays, it’s possible that we could still match this with a light event and correct the event time with the associated t0. I guess this would mostly occur when there is some inefficiency in the pacman receiving the threshold trigger signal from LRS. In principle, this shouldn’t happen, but I seem to remember this happening in the single module runs in some instances. (We should try and catch this in DQM.)
